src/transformStateWithClones.js

```diff
@@ -6,8 +6,29 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
```

Comment:
[DONT REPEAT YOURSELF]: There is no need to create a new copy of currentState at the beginning of the removeProperties case since the object will be modified anyway. You can perform the deletion directly on the currentState.